### PR TITLE
Show embedded entities on the catalog view

### LINKED ISSUE #3214: Support embedded entities for attributes in the catalog
Currently the catalog attribute that does not support embedded entities. The goal o this ticket is to update this view to:

1. Support embedded entities (and later also inter service relations)
2. Cleanup the columns

The new view should:

- Use the nested table view like in the instance attributes.
- Show the description
- Show the type

### PR DESCRIPTION
**Pull request by andraskvr**
# Description

closes #3214 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Code is clear and sufficiently documented
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [x] ~End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )~

# Reviewer Checklist:

- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Code is clear and sufficiently documented
- [ ] Correct, in line with design

### CODE REVIEW

File: src/UI/Components/TreeTable/Catalog/AttributeHelper.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import { AttributeHelper } from "@/UI/Components/TreeTable/Helpers/AttributeHelper";
+import { MultiAttributeNodeDict } from "@/UI/Components/TreeTable/Helpers/AttributeNode";
+import { CatalogAttributeTree } from "@/UI/Components/TreeTable/types";
+
+export class CatalogAttributeHelper
+  implements AttributeHelper<CatalogAttributeTree>
+{
+  constructor(private readonly separator: string) {}
+
+  getPaths(container: CatalogAttributeTree["source"]): string[] {
```
Comment (FloLey) [inline src/UI/Components/TreeTable/Catalog/AttributeHelper.ts:10]: public getPaths?

**Review by FloLey — CHANGES_REQUESTED** [review]
I think this looks good. I like to add 'public' everywhere just for consistency. I find the test really useful and easy to understand and they help to understand the code.

File: src/UI/Components/TreeTable/Catalog/TreeTableHelper.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import { TreeTableHelper } from "@/UI/Components/TreeTable/Helpers/TreeTableHelper";
+import { TreeRow } from "@/UI/Components/TreeTable/TreeRow";
+import {
+  extractCatalogValues,
+  TreeRowCreator,
+} from "@/UI/Components/TreeTable/TreeRow/TreeRowCreator";
+import { CatalogAttributeTree } from "@/UI/Components/TreeTable/types";
+import { words } from "@/UI/words";
+
+export class CatalogTreeTableHelper implements TreeTableHelper {
```
[the remark this answers is not in the document]
Comment (andraskvr) [reply src/UI/Components/TreeTable/Catalog/TreeTableHelper.ts:16]: I'll check again, it's a good suggestion

File: src/Slices/ServiceCatalog/UI/Tabs/AttributeTable.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
```
[the remark this answers is not in the document]
Comment (FloLey) [reply src/Slices/ServiceCatalog/UI/Tabs/AttributeTable.stories.tsx:1]: I didn't know this, this is so cool :p

**Pull request closed without merging**